saleor/graphql/product/dataloaders/products.py

```diff
@@ -548,7 +547,7 @@ def batch_load(self, keys):
             product_collection_map[pid].append(cid)
 
         def map_collections(collections):
-            collection_map = {c.id: c for c in collections}
+            collection_map = {c.id: c for c in collections if c is not None}
```

Comment:
It won't solve the problem. You will change the line where an error can occur. The error is raised as probably the collection was removed in the meantime. In line 552, there is fetch via key, which will raise an exception, when the collection ID doesn't exist - and there is a possibility that the collection will not exist as you added validation for `None`.

Reply:
thanks, I checked the case when a Collection is removed. Now there is no issue with handling that situation.